### ISSUE TITLE
EM throws error on predict2 with 1-d samples

### ISSUE THREAD

**Issue by adiamond** [kyamagu/mexopencv#322]
Error message (shown on matlab command window):
```
Unexpected Standard exception from MEX file.
What() is:
C:\buildslave64\win64_amdocl\master_PackSlave-win64-vc14shared\opencv\modules\ml\src\em.cpp:568:
error: (-215) sample.size() == Size(means.cols, 1) in function
cv::ml::EMImpl::computeProbabilities
```
EM on 1d is overkill but I wanted the posteriors. 

Changing the Properties in `TestEM` to 1D data,  
```matlab
X = [randn(50,1)+1; randn(50,1)-1; randn(50,1)-2];
```
and running the test will yield the error

**Comment by amroamroamro** [comment]
Confirmed. The problem is this part in [`EM_.cpp`](https://github.com/kyamagu/mexopencv/blob/master/src/%2Bcv/private/EM_.cpp#L278):
```cpp
Mat samples(rhs[2].toMat(rhs[2].isSingle() ? CV_32F : CV_64F));
if (samples.rows == 1 || samples.cols == 1)
    samples = samples.reshape(1,1); // ensure 1xd vector if one sample
```
In case the samples are 1D, the matrix will be incorrectly reshaped into a row vector.

*(The reason we manually call `predict2` over each sample is that the method only supports predicting one sample at a time).*

The solution is to drop the reshape line and expect the user to always pass data in row-layout.
Thanks for the bug report, I'll include a fix in the next update.

**Comment by amroamroamro** [comment]
fixed in f690bf0 (part of PR #329).
